hack/verify-spelling.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+export KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# The git commit sha1s here should match the values in $KUBE_ROOT/WORKSPACE.
+kube::util::go_install_from_commit \
+    github.com/client9/misspell/cmd/misspell \
+    9ce5d979ffdaca6385988d7ad1079a33ec942d20
+
+# Spell checking
+# skipping vendor/, translations/, third_party/ and all the other generated files
+git ls-files | grep -v -e 'vendor/' -e 'translations/' -e 'third_party/' -e 'generated' \
+    -e 'staging/src/k8s.io/apiserver/pkg/server/routes/data/swagger/datafile.go' | xargs \
+    misspell -i "Creater,creater" -error
```

Comment:
Leave a comment about why creater is acceptable?

Reply:
Actually this is a legacy.

https://github.com/kubernetes/kubernetes/blob/7223729d51ce19987efb2fb7b1692f75fc43af44/staging/src/k8s.io/apiserver/pkg/endpoints/groupversion.go#L67

Reply:
how about avoiding autogenerated files like `md, man pages, swagger doc` which are getting generated by go source anyways.

Reply:
Adding `swagger` is ok and verified.

How about `grep -v -e 'vendor/' -e 'translations/' -e 'third_party/' -e 'swagger' -e 'docs/man/'`.
@mkumatag @ixdy @liggitt @fejta WDYT?

> though I'm not sure I'd block merge on spell check... for example CHANGELOG files are pushed by the release tool

For `CHANGELOG` files, shall we skip the checking? Actually we could let the release tool automatically fix those typos.